### ISSUE TITLE
Need to update gridding reconstruction for variable stack density

### ISSUE THREAD

**Issue by mathrip** [CEA-COSMIC/pysap-mri#90]
Currently get_stack_fourier and gridded_inverse_fourier_transform_stack are adapted to take as input SoS where each stack is the same trajectory. 
To be able to reconstruct data from SoS with rotation stack or spherical SoS we need to update this function.

This will enable to perform gridding reconstruction with stack and smaps extraction using stack or gridding.

**Comment by mathrip** [comment]
This issue is still occuring.

**Comment by chaithyagr** [comment]
Yes, we hadnt fixed this. Do you want to make this higher priority?

**Comment by mathrip** [comment]
This is not high priority , but it will be great if we fix this soon , just so we have a quick and reliable gridding reconstruction for everything
